core/src/bson_type_info.rs

```diff
@@ -1,214 +1,426 @@
         fixed_bytes_length: None,
+        literal_prefix: None,
+        literal_suffix: None,
+        sql_code: None,
+        is_auto_unique_value: None,
+        is_unsigned: None,
+        num_prec_radix: None,
     };
     pub const UNDEFINED: BsonTypeInfo = BsonTypeInfo {
         type_name: "undefined",
         sql_type: SqlDataType::UNKNOWN_TYPE,
-        searchable: true,
+        non_concise_type: SqlDataType::UNKNOWN_TYPE,
+        searchable: SQL_PRED_BASIC,
+        is_case_sensitive: false,
+        fixed_prec_scale: false,
         scale: None,
         precision: None,
         octet_length: None,
         fixed_bytes_length: None,
+        literal_prefix: None,
+        literal_suffix: None,
+        sql_code: None,
+        is_auto_unique_value: None,
+        is_unsigned: None,
+        num_prec_radix: None,
     };
     pub const OBJECTID: BsonTypeInfo = BsonTypeInfo {
         type_name: "objectId",
         sql_type: SqlDataType::UNKNOWN_TYPE,
-        searchable: true,
+        non_concise_type: SqlDataType::UNKNOWN_TYPE,
+        searchable: SQL_PRED_BASIC,
+        is_case_sensitive: false,
+        fixed_prec_scale: false,
         scale: None,
         precision: Some(24),
         octet_length: Some(24),
         fixed_bytes_length: None,
+        literal_prefix: None,
+        literal_suffix: None,
+        sql_code: None,
+        is_auto_unique_value: Some(true),
+        is_unsigned: None,
+        num_prec_radix: None,
     };
     pub const BOOL: BsonTypeInfo = BsonTypeInfo {
         type_name: "bool",
         sql_type: SqlDataType::EXT_BIT,
-        searchable: true,
+        non_concise_type: SqlDataType::EXT_BIT,
+        searchable: SQL_PRED_BASIC,
+        is_case_sensitive: false,
+        fixed_prec_scale: false,
         scale: None,
         precision: Some(1),
         octet_length: Some(1),
         fixed_bytes_length: Some(1),
+        literal_prefix: None,
+        literal_suffix: None,
+        sql_code: None,
+        is_auto_unique_value: None,
+        is_unsigned: None,
+        num_prec_radix: None,
     };
     pub const DATE: BsonTypeInfo = BsonTypeInfo {
         type_name: "date",
         sql_type: SqlDataType::TIMESTAMP,
-        searchable: true,
+        non_concise_type: SqlDataType::DATETIME,
+        searchable: SQL_PRED_BASIC,
+        is_case_sensitive: false,
+        fixed_prec_scale: true,
         scale: Some(3),
         precision: Some(23),
         octet_length: Some(16),
         fixed_bytes_length: Some(16),
+        literal_prefix: Some("'"),
+        literal_suffix: Some("'"),
+        sql_code: Some(3),
+        is_auto_unique_value: None,
+        is_unsigned: None,
+        num_prec_radix: None,
     };
     pub const NULL: BsonTypeInfo = BsonTypeInfo {
         type_name: "null",
         sql_type: SqlDataType::UNKNOWN_TYPE,
-        searchable: true,
+        non_concise_type: SqlDataType::UNKNOWN_TYPE,
+        searchable: SQL_PRED_BASIC,
```

Comment:
this is a good call out, and I believe nulls _cannot_ be compared, if I am understanding [these spec tests](https://github.com/10gen/mongosql-rs/blob/0cc4cc824a1eeab5cf4fc568bb5fa26a075e2dbc/tests/spec_tests/query_tests/operators_comparison.yml#L958) correctly, this may be better as `SQL_PRED_NONE`

Reply:
You are understanding those tests correctly. In general, when either operand of a comparison operator is NULL or MISSING, the result is NULL.